python/cugraph/cugraph/dask/community/ktruss_subgraph.py

```diff
@@ -0,0 +1,116 @@
+# Copyright (c) 2022-2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from dask.distributed import wait, default_client
+
+import cugraph.dask.comms.comms as Comms
+import dask_cudf
+import cudf
+import cupy as cp
+from typing import Tuple
+
+from pylibcugraph import ResourceHandle, k_truss_subgraph as pylibcugraph_k_truss_subgraph
+
+
+def _call_k_truss_subgraph(
+    sID: bytes,
+    mg_graph_x,
+    k: int,
+    do_expensive_check: bool,
+) -> Tuple[cp.ndarray, cp.ndarray, cp.ndarray]:
+
+    return pylibcugraph_k_truss_subgraph(
+        resource_handle=ResourceHandle(Comms.get_handle(sID).getHandle()),
+        graph=mg_graph_x,
+        k=k,
+        do_expensive_check=do_expensive_check,
+    )
+
+
+def convert_to_cudf(cp_arrays: cp.ndarray) -> cudf.DataFrame:
+    cp_src, cp_dst, cp_weight = cp_arrays
+
+    df = cudf.DataFrame()
+    if cp_src is not None:
+        df["src"] = cp_src
+        df["dst"] = cp_dst
+    if cp_weight is not None:
+        df["weight"] = cp_weight
+
+    return df
+
+
+def k_truss_subgraph(input_graph, k: int) -> dask_cudf.DataFrame:
+    """
+    Returns the K-Truss subgraph of a graph for a specific k.
+
+    The k-truss of a graph is a subgraph where each edge is part of at least
+    (k−2) triangles. K-trusses are used for finding tighlty knit groups of
+    vertices in a graph. A k-truss is a relaxation of a k-clique in the graph
+    and was define in [1]. Finding cliques is computationally demanding and
+    finding the maximal k-clique is known to be NP-Hard.
```

Comment:
it says "and was define in [1]" but I don't see a citation anywhere for [1].  If that can't be included, then maybe it shouldn't be mentioned:
```suggestion
    vertices in a graph. A k-truss is a relaxation of a k-clique in the
    graph. Finding cliques is computationally demanding and finding
    the maximal k-clique is known to be NP-Hard.
```